src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -248,8 +256,22 @@ protected override AuthenticationTicket AuthenticateCore()
 
             OpenIdConnectMessage message = null;
 
+            if (string.Equals(Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
+            {
+                message = new OpenIdConnectMessage(Request.Query);
+
+                // response_mode=query (explicit or not) and a response_type containing id_token
+                // or token are not considered as a safe combination and MUST be rejected.
+                // See http://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Security
+                if (!string.IsNullOrWhiteSpace(message.IdToken) || !string.IsNullOrWhiteSpace(message.Token))
+                {
+                    Logger.LogError("An OpenID Connect response cannot contain an identity token " +
```

Comment:
I'm pretty sure we don't localize log messages. They are currently intended to be English only because there's no way to do localized logs. The problem with having localized log messages is which culture do you use for them? Surely you don't want to use the current thread's culture, because that is usually set to whatever the *end user's* culture is. So if a French user visits, I get French logs. And the next user is German, so the next log is German. Clearly that is not going to work. So in the meantime we decided to not localize them.